### ISSUE TITLE
start-single-node: Setting timeseries.storage.resolution_10s.ttl = '0s' does not delete existing logs.

### ISSUE THREAD

**Issue by cubuzz** [cockroachdb/cockroach#61669]
**Describe the problem**

It appears when running a single node cluster, `SET CLUSTER SETTING timeseries.storage.resolution_10s.ttl = '0s';` will not delete existing time series data.

**To Reproduce**

1. Set up CockroachDB node, start it with `start-single-node`
2. Record Time Series for a while
3. Disable Time Series and `SET CLUSTER SETTING timeseries.storage.resolution_10s.ttl = '0s';`
4. Check back in a few hours / days

**Expected behaviour**
All timeseries data has been deleted.

**Actual behaviour**
Timeseries data persists.

**Environment:**
 - CockroachDB 20.0.2.5
 - Server OS: [e.g. Ubuntu Server 20.04 LTS]

**Additional context**
A lot of storage space is wasted by unnecessary time series data. 

I assume some other garbage collections don't run properly either. However, so far I've been unable to pinpoint which ones exactly are wasting hard drive space.

**Comment by andreimatei** [comment]
Thanks for the report; we'll take a look.

**Comment by rimadeodhar** [comment]
So I took a look at the final outstanding problem here which is to check why does the DB console UI continue to show timeseries storage size as a non-zero value even after all timeseries data is deleted. The summary of this investigation is as follows:
1. To estimate the disk usage for timeseries info, we call into pebble DBs [estimateDiskUsageInBytes](https://github.com/cockroachdb/pebble/blob/master/db.go) method which returns a non zero value.
2. The reason for this is because when data is deleted from Pebble, we don't actually delete the data but write a tombstone. The data is actually deleted lazily during compactions. Pebble uses some statistics about deletions to confirm which compactions to perform, so it may be a while before this data is physically deleted. Pebble ensures that the amount of wasted space is bounded, so this is not a problem.
3. In addition, the EstimateDiskUsageInBytes method uses the granularity of data blocks to measure disk usage so it will always return the size of atleast one block as disk usage even if there are no keys in a range.

Thanks @jbowens for explaining Pebble's disk usage estimation logic.

So to summarize, in order to delete timeseries data, you should set timeseries ttl cluster settings for 10s and 30m resolution to 0s and disable timeseries. This will ensure that all timeseries data is deleted in ~24 hours. The disk usage reflected for timeseries within the DB console may take longer to drop due to Pebble's compaction algorithm but that isn't an issue as we bound the amount of wasted space to ensure we are not truly wasting disk space beyond a certain limit. 

I have updated cockroach docs to reflect on how to delete all timeseries cluster data:. https://www.cockroachlabs.com/docs/v20.2/operational-faqs.html. Closing out this issue since all related investigations are resolved.
